engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -120,11 +123,15 @@ private void createFBOs() {
         }
     }
 
-    // TODO: Pairing FBOs for swapping functionality
     public void swapReadWriteBuffers() {
-        FBO fbo = get(READONLY_GBUFFER);
-        fboLookup.put(READONLY_GBUFFER, get(WRITEONLY_GBUFFER));
-        fboLookup.put(WRITEONLY_GBUFFER, fbo);
-        notifySubscribers();
+        buffersSwapped = !buffersSwapped;
```

Comment:
I still have some concerns about this, i.e. the .swap() method called in the constructor of nodes rather than in process. But overall I think I understand this better and it makes sense.